backend/LfClassicData/SystemDbContext.cs

```diff
@@ -17,4 +19,16 @@ public SystemDbContext(MongoClient mongoClient)
 
     internal IMongoCollection<User> Users { get; }
     public IMongoCollection<LfProject> Projects { get; }
+
+    public async Task<bool> IsAvailable()
+    {
+        if (_isAvailable is null or false)
+        {
+            _isAvailable = await Task.Run(() =>
+            {
+                return _mongoDatabase.RunCommandAsync((Command<BsonDocument>)"{ping:1}").Wait(100);
+            });
```

Comment:
No, `Wait(100)` never throws. It returns a boolean indicating whether the command finished within the timeout.

Reply:
ahh ok, I forget some of them throw on timeout. It would still probably be better to catch exceptions and not use Wait as that's intended as a last resort when calling an Async method from a sync context. I'd use WaitAsync for that instead